From d30f834fcec7ecdbb74e263789db2fbe9ff37e99 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Fri, 11 Oct 2024 16:12:50 +0200
Subject: [PATCH] Skip abi3 tests when freethreading Python build is used

Skip the tests using abi3 extension when the freethreading build is
used, as Python 3.13 freethreading does not support the limited API
and its use results in a build error.

Fixes #636
---
 tests/conftest.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/conftest.py b/tests/conftest.py
index 5ee84da..d5bd356 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -7,6 +7,7 @@ from __future__ import annotations
 import os.path
 import subprocess
 import sys
+import sysconfig
 
 import pytest
 
@@ -22,7 +23,7 @@ def wheels_and_eggs(tmp_path_factory):
         "unicode.dist",
     )
 
-    if sys.platform != "win32":
+    if sys.platform != "win32" and sysconfig.get_config_var("Py_GIL_DISABLED") != 1:
         # ABI3 extensions don't really work on Windows
         test_distributions += ("abi3extension.dist",)
 
-- 
2.47.0

